File: 0001-two-sum/0001-two-sum.py
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        substraction = dict()

        for i in range(len(nums)):
            if nums[i] in substraction:
                return [substraction[nums[i]], i]
            else:
                substraction[target - nums[i]] = i
            

File: 0001-two-sum/0001-two-sum.go
func twoSum(nums []int, target int) []int {
    diffs := make(map[int]int)

    for i1, n := range nums {
        i2, exists := diffs[n]
        if exists {
            return []int{i1, i2}
        }
        diffs[target - n] = i1
    }

    return nil
}